src/controllers/expenseController.js

```diff
@@ -0,0 +1,168 @@
+const expenseService = require('../services/expenseService');
+const { getUserById } = require('../services/userService');
+const { STATUS_CODE } = require('../utils/statusCodes');
+
+const getAll = (req, res) => {
+  const { userId, categories, from, to } = req.query;
+  const id = Number(userId);
+
+  const AllExpenses = expenseService.getAllExpenses(id, categories, from, to);
+
+  res.send(AllExpenses);
+};
+
+const create = (req, res) => {
+  const { userId, spentAt, title, amount, category, note } = req.body;
+  const user = getUserById(userId);
+
+  if (!user) {
+    res.sendStatus(STATUS_CODE.NOT_FOUND);
+
+    return;
+  }
+
+  if (!userId || typeof userId !== 'number') {
+    res.sendStatus(STATUS_CODE.NOT_FOUND);
+
+    return;
+  }
+
+  if (!spentAt || isNaN(new Date(spentAt))) {
+    res.sendStatus(STATUS_CODE.NOT_FOUND);
+
+    return;
+  }
+
+  if (!title || typeof title !== 'string') {
+    res.sendStatus(STATUS_CODE.NOT_FOUND);
+
+    return;
+  }
+
+  if (!amount || typeof amount !== 'number') {
+    res.sendStatus(STATUS_CODE.NOT_FOUND);
+
+    return;
+  }
+
+  if (!category || typeof category !== 'string') {
+    res.sendStatus(STATUS_CODE.NOT_FOUND);
+
+    return;
+  }
+
+  if (!note || typeof note !== 'string') {
+    res.sendStatus(STATUS_CODE.NOT_FOUND);
+
+    return;
+  }
```

Comment:
To handle multiple conditions we basically use switch/case.

Also, you can try to create some variable like `isValidBody` and write just one if statement

Reply:
I didn't catch how can I use switch/case here, but made a variable.